contracts/Relayer.sol

```diff
@@ -63,89 +56,73 @@ contract Relayer is IRelayer {
         return _relayers[relayerAddress];
     }
 
-    function getRelayerThreshold() public view returns (uint) {
-        return _relayerThreshold;
-    }
-
     function getTotalRelayers() public override returns (uint) {
         return _totalRelayers;
     }
 
     function getCurrentRelayerThresholdProposal() public view returns (
-        uint, uint, uint, string memory) {
+        uint, address[] memory, address[] memory, VoteStatus) {
         return (
         _currentRelayerThresholdProposal._proposedValue,
-        _currentRelayerThresholdProposal._numYes,
-        _currentRelayerThresholdProposal._numNo,
-        _voteStatusStrings[uint(_currentRelayerThresholdProposal._status)]);
+        _currentRelayerThresholdProposal._yesVotes,
+        _currentRelayerThresholdProposal._noVotes,
+        _currentRelayerThresholdProposal._status);
     }
 
     function getRelayerProposal(address proposedAddress) public view returns (
-        address, string memory, uint, uint, string memory) {
+        RelayerActionType, address[] memory, address[] memory, VoteStatus) {
         RelayerProposal memory relayerProposal = _relayerProposals[proposedAddress];
         return (
-        relayerProposal._proposedAddress,
-        _relayerActionTypeStrings[uint(relayerProposal._action)],
-        relayerProposal._numYes,
-        relayerProposal._numNo,
-        _voteStatusStrings[uint(relayerProposal._status)]);
+            relayerProposal._action,
+            relayerProposal._yesVotes,
+            relayerProposal._noVotes,
+            relayerProposal._status);
     }
 
-    function createRelayerProposal(address proposedAddress, RelayerActionType action) public override _onlyRelayers {
-        require(uint(action) <= 1, "action out of the vote enum range");
-        require(action == RelayerActionType.Remove && _relayers[proposedAddress] == true, "address is not a relayer");
-        require(action == RelayerActionType.Add && _relayers[proposedAddress] == false, "address is currently a relayer");
-        require(_relayerProposals[proposedAddress]._status == VoteStatus.Inactive, "there is already an active proposal for this address");
-
-        _relayerProposals[proposedAddress] = RelayerProposal({
-            _proposedAddress: proposedAddress,
-            _action: action,
-            _numYes: 1, // Creator always votes in favour
-            _numNo: 0,
-            _status: VoteStatus.Active
-            });
+    function voteRelayerProposal(address proposedAddress, RelayerActionType action) public override _onlyRelayers {
```

Comment:
This looks like the proposal can be stuck, is there a reason we can't vote against?

Reply:
The proposal can get stuck, and I can certainly add the ability to vote against it, but I was also thinking that even if the proposal does get stuck, it doesn't really matter. If an address is not a relayer, the only action a proposal could have is to add them, so voting against the proposal and making it inactive just means a relayer has to reopen the proposal v.s. just having the proposal be active until the yes votes surpass the threshold.

I initially wrote this method without the vote just to match how the deposit proposals in `Bridge.sol` ignore no votes for now, but if the above explanation seems problematic in any way, it shouldn't be difficult to add the vote back in

Reply:
Hmm ok I can actually see the logic of this.